### PR TITLE
[AutoPR authorization/resource-manager] All updates

### PR DESCRIPTION
**Pull request by lmazuel**
@yugangw-msft Ready to install as 0.40.0

### CODE REVIEW

**Comment by lmazuel** [comment]
Tests KO since 2018-01-01-preview is not out yet.

**Comment by lmazuel** [comment]
@AutorestCI rebase

**Comment by lmazuel** [comment]
@AutorestCI rebuild https://github.com/Azure/azure-rest-api-specs/blob/master/specification/authorization/resource-manager/readme.md

**Comment by darshanhs90** [comment]
> 

you might need to rerecord the tests,since this was recorded before the new api version was rolled out

---
Refers to: azure-mgmt-authorization/tests/recordings/test_mgmt_authorization.test_authorization.yaml:29 in 62edb5d. [](commit_id = 62edb5db7a90764cb7b4e1fb2a02f8a2fed6ecdf, deletion_comment = False)

**Comment by darshanhs90** [comment]
The api version is out now.

---
In reply to: [367439171](https://github.com/Azure/azure-sdk-for-python/pull/1916#issuecomment-367439171) [](ancestors = 367439171)

**Comment by lmazuel** [comment]
@yugangw-msft for review as well

**Comment by yugangw-msft** [comment]
@lmazuel , I have verified, could you please publish it?

File: azure-mgmt-authorization/azure/mgmt/authorization/models/classic_administrator.py [hunk 1]
```diff
@@ -21,20 +21,24 @@ class ClassicAdministrator(Model):
...
+    :param role: The role of the administrator.
+    :type role: str
     """
 
     _attribute_map = {
         'id': {'key': 'id', 'type': 'str'},
         'name': {'key': 'name', 'type': 'str'},
         'type': {'key': 'type', 'type': 'str'},
-        'properties': {'key': 'properties', 'type': 'ClassicAdministratorProperties'},
+        'email_address': {'key': 'properties.emailAddress', 'type': 'str'},
```
Comment (lmazuel) [inline azure-mgmt-authorization/azure/mgmt/authorization/models/classic_administrator.py]: breaking:
properties.email_address => email_address
properties.role => role

File: azure-mgmt-authorization/azure/mgmt/authorization/models/permission.py [hunk 2]
```diff
@@ -19,13 +19,22 @@ class Permission(Model):
...
     }
 
-    def __init__(self, actions=None, not_actions=None):
-        self.actions = actions
-        self.not_actions = not_actions
+    def __init__(self, **kwargs):
+        super(Permission, self).__init__(**kwargs)
+        self.actions = kwargs.get('actions', None)
+        self.not_actions = kwargs.get('not_actions', None)
+        self.data_actions = kwargs.get('data_actions', None)
```
Comment (lmazuel) [inline azure-mgmt-authorization/azure/mgmt/authorization/models/permission.py]: new data_actions / not_data_actions

File: azure-mgmt-authorization/azure/mgmt/authorization/models/provider_operation.py [hunk 3]
```diff
@@ -25,6 +25,8 @@ class ProviderOperation(Model):
     :type origin: str
     :param properties: The operation properties.
     :type properties: object
+    :param is_data_action: The dataAction flag to specify the operation type.
```
Comment (lmazuel) [inline azure-mgmt-authorization/azure/mgmt/authorization/models/provider_operation.py]: new is_data_action

File: azure-mgmt-authorization/azure/mgmt/authorization/models/role_assignment.py [hunk 4]
```diff
@@ -15,26 +15,47 @@
...
+        'name': {'readonly': True},
+        'type': {'readonly': True},
+    }
+
     _attribute_map = {
         'id': {'key': 'id', 'type': 'str'},
         'name': {'key': 'name', 'type': 'str'},
         'type': {'key': 'type', 'type': 'str'},
-        'properties': {'key': 'properties', 'type': 'RoleAssignmentPropertiesWithScope'},
+        'scope': {'key': 'properties.scope', 'type': 'str'},
```
Comment (lmazuel) [inline azure-mgmt-authorization/azure/mgmt/authorization/models/role_assignment.py]: flattening

File: azure-mgmt-authorization/azure/mgmt/authorization/models/role_assignment_create_parameters.py [hunk 5]
```diff
@@ -15,14 +15,26 @@
...
+     principal, or security group.
+    :type principal_id: str
+    :param can_delegate: The delgation flag used for creating a role
+     assignment
+    :type can_delegate: bool
     """
 
     _attribute_map = {
-        'properties': {'key': 'properties', 'type': 'RoleAssignmentProperties'},
+        'role_definition_id': {'key': 'properties.roleDefinitionId', 'type': 'str'},
```
Comment (lmazuel) [inline azure-mgmt-authorization/azure/mgmt/authorization/models/role_assignment_create_parameters.py]: flattening

File: azure-mgmt-authorization/azure/mgmt/authorization/operations/permissions_operations.py [hunk 6]
```diff
@@ -22,25 +22,26 @@ class PermissionsOperations(object):
     :param client: Client for service requests.
     :param config: Configuration of service client.
     :param serializer: An object model serializer.
-    :param deserializer: An objec model deserializer.
-    :ivar api_version: The API version to use for this operation. Constant value: "2015-07-01".
+    :param deserializer: An object model deserializer.
+    :ivar api_version: The API version to use for this operation. Constant value: "2018-01-01-preview".
```
Comment (lmazuel) [inline azure-mgmt-authorization/azure/mgmt/authorization/operations/permissions_operations.py]: new api version

File: azure-mgmt-authorization/azure/mgmt/authorization/operations/role_assignments_operations.py [hunk 7]
```diff
@@ -455,21 +455,17 @@ def delete_by_id(
             return client_raw_response
 
         return deserialized
+    delete_by_id.metadata = {'url': '/{roleId}'}
 
     def create_by_id(
-            self, role_assignment_id, properties=None, custom_headers=None, raw=False, **operation_config):
+            self, role_id, parameters, custom_headers=None, raw=False, **operation_config):
```
Comment (lmazuel) [inline azure-mgmt-authorization/azure/mgmt/authorization/operations/role_assignments_operations.py]: new name

**Pull request merged**